### PR TITLE
chore: replace `expect` calls

### PR DESCRIPTION
**Pull request by TropicalDog17**

### CODE REVIEW

File: near-sdk/src/environment/env.rs
Comment (frol) [inline near-sdk/src/environment/env.rs:467]: Use `panic_str` here (it is actually defined in this file)

File: near-sdk/src/environment/env.rs
Comment (frol) [inline near-sdk/src/environment/env.rs:483]: `panic_str`

File: near-sdk-macros/src/core_impl/code_generator/impl_item_method_info.rs
Comment (frol) [inline near-sdk-macros/src/core_impl/code_generator/impl_item_method_info.rs:126]: Please, use fully-qualified syntax here (add `::` in the beginning) since this code is generated into the user code and we want to avoid an unexpected aliasing:

```suggestion
                        None => ::near_sdk::env::panic_str("Expected input since method has arguments.")
```

P.S. Apply this change in other places where applicable (in near-sdk-marcos)

**Review by frol — CHANGES_REQUESTED** [review]
4kb is already not too bad. I have spotted a few more potential improvements that could further drop the size. I believe that if we manage to avoid all instances of string formatting (`.expect` uses formatting under the hood to include the source file name and line of code), we could see a major drop in size, but for that someone would need to check the produced Wasm file static strings and identify how we can avoid those.

As a rule of thumb, `panic_any` should not be used here. We should either use `near_sdk::env::panic_str` or otherwise it is an indicator that `.expect` is not going to be part of Wasm file anyway.

**Comment by TropicalDog17** [comment]
Hey I've applied based on your review, and also further optimize in ([8296a73](https://github.com/near/near-sdk-rs/pull/1220/commits/8296a73075094812002a313d75711d82c8ee4fb6)), the result is more significant on macros/decorators-heavy contract like `examples/adder`, the code size is reduce from 78.0kB to 69,3kB, which is an improvement of about ~11% .
image

image

File: near-sdk-macros/src/core_impl/code_generator/attr_sig_info.rs
Comment (TropicalDog17) [reply near-sdk-macros/src/core_impl/code_generator/attr_sig_info.rs:257]: yeah I removed it accidentally, I will revert

File: near-sdk-macros/src/core_impl/code_generator/attr_sig_info.rs
Comment (TropicalDog17) [reply near-sdk-macros/src/core_impl/code_generator/attr_sig_info.rs:217]: yeah let aside the bug you mentioned,  removing the format! macro does reduce the code size slightly according to my testing. I will update the metrics asap.

**Comment by TropicalDog17** [comment]
hey @frol here is the report, let me know if the pr need further fix
| Contract | Original Size (B) | [bd1dbba](https://github.com/near/near-sdk-rs/pull/1220/commits/bd1dbba9320f0e2b1318ac15e3bb5d01544e380b) (B) | Improvement (%) | [8405364](https://github.com/near/near-sdk-rs/pull/1220/commits/8405364f7a2107507793b4141b6a26eb8bffa03d) - latest (B) | Improvement (%) |
|----------|-------------------|--------------------|-----------------|-----------------------|-----------------|
| adder | 78039 | 69818 | 10.53% | 69333 | 11.15% |
| callback-results | 84004 | 79800 | 5.00% | 79800 | 5.00% |
| cross-contract-calls (high) | 80203 | 72518 | 9.58% | 72518 | 9.58% |
| cross-contract-calls (low) | 74003 | 69792 | 5.69% | 69792 | 5.69% |
| factory-contract (high) | 179885 | 167965 | 6.63% | 167965 | 6.63% |
| factory-contract (low) | 173529 | 165336 | 4.72% | 165336 | 4.72% |
| fungible-token (defi) | 99552 | 91987 | 7.60% | 91987 | 7.60% |
| fungible-token | 165376 | 160969 | 2.66% | 160969 | 2.66% |
| lockable-fungible-token | 120367 | 111769 | 7.14% | 111769 | 7.14% |
| mission-control | 116445 | 108107 | 7.16% | 108107 | 7.16% |
| non-fungible-token (approval) | 96835 | 88415 | 8.70% | 88415 | 8.70% |
| non-fungible-token | 228686 | 221936 | 2.95% | 221936 | 2.95% |
| non-fungible-token (receiver) | 96576 | 88150 | 8.72% | 88150 | 8.72% |
| status-message | 87770 | 83848 | 4.47% | 83848 | 4.47% |
| test-contract | 26688 | 22490 | 15.73% | 22490 | 15.73% |
| versioned | 101857 | 95806 | 5.94% | 95806 | 5.94% |

File: near-sdk-macros/src/core_impl/code_generator/snapshots/callback_args_vec.snap
Comment (frol) [inline near-sdk-macros/src/core_impl/code_generator/snapshots/callback_args_vec.snap:40]: `stringify!(i)` will return `"i"` instead of the value under `i` variable, which means that the error will be `Callback computation i was not successful`, instead of `Callback computation 0 was not successful`.

I think `format!` is unavoidable here.

**Review by frol — CHANGES_REQUESTED** [review]
@TropicalDog17 Please, address the review comment and fix the issues reported by CI.

**Comment by TropicalDog17** [comment]
@frol I've fixed based on your review. About the failing CI it seems it also fails on `master` branch. I don't have much experience in github actions/CI, so I guess it should be addressed in a seperate PR.

**Comment by frol** [comment]
Yeah, I created an issue for that: #1223

**Comment by akorchyn** [comment]
@TropicalDog17 Sorry for jumping in, I'd like to invite you to participate in the [Race of Sloths](https://race-of-sloths.com/). Just mention @race-of-sloths user in your GitHub comment or PR description to join!

See how the flow works here: https://github.com/NEAR-DevHub/neardevhub-bos/pull/910

**Comment by TropicalDog17** [comment]
@race-of-sloths

**Comment by race-of-sloths** [comment]
@TropicalDog17 Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
New Sloth joined the Race! Welcome!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/TropicalDog17?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/TropicalDog17?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/TropicalDog17)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command
</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/TropicalDog17)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Comment by race-of-sloths** [comment]
### 🔄 The PR has been merged.
>[!IMPORTANT]
This pull request is a part of the Race of Sloths and has not been scored yet. Scoring will close in 24 hours! 🕰️

**Comment by frol** [comment]
@race-of-sloths score 3

**Comment by race-of-sloths** [comment]
### 🌟 Score recorded!

@frol, thank you for scoring this pull request in the Race of Sloths!

**Comment by race-of-sloths** [comment]
### ✅ PR is finalized! 
Your contribution is much appreciated with a final score of 3!
You have received 40 (30 base + 10 weekly bonus) Sloth points for this contribution


Another weekly streak completed, well done @TropicalDog17! To keep your weekly streak and get another bonus make pull request next week! Looking forward to see you in race-of-sloths

**Pull request merged**
